macros/src/lib.rs

```diff
@@ -458,3 +461,140 @@ impl Parse for OptionSetter {
         Ok(Self { attrs, name, type_ })
     }
 }
+
+#[import_tokens_attr]
```

Comment:
The core magic that the `macro_magic` crate gives us is that while this proc macro is attached to an `impl` block that it edits to add the action option setters, it _also_ receives the full "exported" token stream for the option struct as its `attr` argument (instead of just the path as it would seem from the call site).

This means that we can remove _all_ of the duplicated information we had to have in `option_setters!` invocations (names, types, cfg attrs) _and_ bring forward more information (doc comments).